src/_cffi_src/openssl/hmac.py

```diff
@@ -80,4 +79,37 @@
         return 0;
 #endif
 }
+
+HMAC_CTX *Cryptography_HMAC_CTX_new(void) {
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L && !defined(LIBRESSL_VERSION_NUMBER)
+    return HMAC_CTX_new();
+#else
+    /* This uses OPENSSL_zalloc in 1.1.0, which is malloc + memset */
+    HMAC_CTX *ctx = (HMAC_CTX *)OPENSSL_malloc(sizeof(HMAC_CTX));
+    memset(ctx, 0, sizeof(HMAC_CTX));
+    /*if (ctx)
+        if (!HMAC_CTX_reset(ctx)) {
+            HMAC_CTX_free(ctx);
+            ctx = NULL;
+        }*/
```

Comment:
Oh, nevermind. I'm less dumb than I thought (but still dumb for leaving this here). So in 1.1.0 HMAC_CTX is opaque and most of the struct members are pointers:
```c
struct hmac_ctx_st {
    const EVP_MD *md;
    EVP_MD_CTX *md_ctx;
    EVP_MD_CTX *i_ctx;
    EVP_MD_CTX *o_ctx;
    unsigned int key_length;
    unsigned char key[HMAC_MAX_MD_CBLOCK];
};
```

In 1.0.2 and below they are directly in the HMAC_CTX struct:
```c
typedef struct hmac_ctx_st
    {
    const EVP_MD *md;
    EVP_MD_CTX md_ctx;
    EVP_MD_CTX i_ctx;
    EVP_MD_CTX o_ctx;
    unsigned int key_length;
    unsigned char key[HMAC_MAX_MD_CBLOCK];
    } HMAC_CTX;
```

So to initialize an HMAC_CTX in 1.1.0 you need to call that HMAC_CTX_reset function, but in 1.0.2 and below just memsetting the allocated memory to zero is sufficient before calling the init functions. So the right thing to do is just delete the commented out lines, as they serve no purpose in older OpenSSL.